elastic-buffer-sim/src/Bittide/Topology/TH.hs

```diff
@@ -137,6 +139,52 @@ plotEbsQ g = do
 cross :: [a] -> [b] -> [(a, b)]
 cross xs ys = (,) <$> xs <*> ys
 
+-- | Successive application on a stream.
+succApp :: (a -> a -> b) -> [a] -> [b]
+succApp op (x:xs@(y:_)) = x `op` y : succApp op xs
+succApp _ _ = error "This function is meant to be called on streams."
+
+streamSettle :: (Ord a, Integral a, Show a) => a -> a -> Bool
+streamSettle x y = asInteger x - asInteger y == 0
+ where
+  asInteger n = fromIntegral n :: Integer
```

Comment:
I think we can make this a bit more configurable. They don't have to be _exactly_ the same (in practice they're never going to be), just within ~5 ppm or so. This should make running the check way quicker too.

For example, if I run `complete3` for 10_000_000 cycles / 8 minutes locally I get:

[clockscomplete3.pdf](https://github.com/bittide/bittide-hardware/files/9989661/clockscomplete3.pdf)
[elasticbufferscomplete3.pdf](https://github.com/bittide/bittide-hardware/files/9989663/elasticbufferscomplete3.pdf)

Which seems (almost?) good enough.